pkg/migration/controllers/migration.go

```diff
@@ -1547,18 +1567,55 @@ func (m *MigrationController) prepareCRDClusterResource(
 					return fmt.Errorf("unable to find suspend path, err: %v", err)
 				}
 				disableVersion = int64(0)
+				if curr == 0 && suspendAnnotationIsPresent {
+					//suspendAnnotation has value set as {currVal + "," + suspend.Value}
+					//we need to extract only the currVal from it
+					annotationValue := strings.Split(SuspendAnnotationValue, ",")[0]
```

Comment:
Is it really required to do the split and again modify the `curr` value and later again add it to the annotation?

If curr == 0 &&  suspendAnnotationIsPresent 
       We don't have to set anything, we can take the object with the annotation itself.

Reply:
1. The condition I am checking as of now is curr == 0 && suspendAnnotation is > 0. So I need to extract the annotation value for that.
2. Additionally I am also updating the suspend path value with the annotationValue in line #1583
This is done for the scenario when migration.spec.startApplications is true.

Let's consider the case where resources were migrated from C1 to C2. Now the resource in C2 is scaled down so actual replica value will be 0 while annotation will have the replicaCount from C1 (let's say it is 3)
Now if I migrate the resource from C2 to C3 with startApplications as true, I would not be scaling down the resource and migrating it as is. For this case I should have modified the actual replica value in the resource as well. This is what is done in #1583

So that in #1614 when we return nil because startApplications is true and we do not need to scale down, the resource applied in C3 will have replicaCount set to 3.

Reply:
Got it thanks, this is required for `startApplications : true`. I did not think about that.